File: src/lib/groq.ts
export const askGroq = async (prompt: string, apiKey: string) => {
  try {
    const response = await fetch('https://api.groq.com/openai/v1/chat/completions', {
      method: 'POST',
      headers: {
        'Authorization': `Bearer ${apiKey}`,
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        messages: [{ role: 'user', content: prompt }],
        model: 'llama-3.3-70b-versatile',
      }),
    });

    const data = await response.json();
    return data.choices[0]?.message?.content || 'No response from Groq.';
  } catch (error) {
    console.error('Groq Error:', error);
    return 'Failed to generate response. Please check your Groq API key.';
  }
};

File: backend/llm/groq.py
"""ORYX LLM — Groq provider (fast cloud inference)."""

import httpx
from backend.config import settings


class GroqProvider:
    """Chat via the Groq API."""

    def __init__(self):
        self.api_key = settings.GROQ_API_KEY
        self.model = settings.GROQ_MODEL
        self.base_url = "https://api.groq.com/openai/v1/chat/completions"

    async def chat(self, messages: list[dict], **kwargs) -> str:
        payload = {
            "model": kwargs.get("model", self.model),
            "messages": messages,
            "temperature": kwargs.get("temperature", 0.7),
            "max_tokens": kwargs.get("max_tokens", 2048),
        }
        headers = {
            "Authorization": f"Bearer {self.api_key}",
            "Content-Type": "application/json",
        }
        async with httpx.AsyncClient(timeout=60) as client:
            resp = await client.post(self.base_url, json=payload, headers=headers)
            resp.raise_for_status()
            return resp.json()["choices"][0]["message"]["content"]

    async def is_available(self) -> bool:
        return bool(self.api_key and self.api_key != "your_groq_api_key_here")
